tests/integration/test_feedback.py

```diff
@@ -69,8 +75,44 @@ def test_feedback_wrong_conversation_id():
     """Check if feedback with wrong conversation ID is not accepted by the service."""
     response = client.post(
         "/v1/feedback",
-        json={"conversation_id": 0, "feedback_object": "blah"},
+        json={"conversation_id": 0, "feedback_object": {"some": "data"}},
     )
     # for the request send w/o proper payload, the server
     # should respond with proper error code
     assert response.status_code == requests.codes.unprocessable_entity
+
+
+def test_feedback_list():
+    """Check if feedback list is returned."""
+    # store some feedback first
+    conversation_id = suid.get_suid()
+    response = client.post(
+        "/v1/feedback",
+        json={"conversation_id": conversation_id, "feedback_object": {"some": "data"}},
+    )
+    assert response.status_code == requests.codes.ok
+
+    response = client.get("/v1/feedback/list")
+
+    assert response.status_code == requests.codes.ok
+    assert response.json() == {"response": str([conversation_id])}
+
+
+def test_feedback_remove():
+    """Check if feedback is removed."""
+    # store some feedback first
+    conversation_id = suid.get_suid()
+    response = client.post(
+        "/v1/feedback",
+        json={"conversation_id": conversation_id, "feedback_object": {"some": "data"}},
+    )
+    assert response.status_code == requests.codes.ok
+
+    # The TestClient.delete() method from Starlette's TestClient does
+    # not accept a json argument - workaround it by using `requests`
+    response = client.request(
+        "DELETE", "/v1/feedback/", json={"feedback_id": conversation_id}
+    )
+
+    assert response.status_code == requests.codes.ok
+    assert response.json() == {"response": "feedback removed"}
```

Comment:
This comment is not related to the line, I just putting it here to have the "threading" ability on the comment.

I didn't implement the tests like `test_feedback_no_payload_send`. Those are basically testing the FastAPI/pydantic behavior. Unless there is a special logic for handling/validating the input parameters, I'm not sure what is the actual benefit of such tests. What we can do if it starts failing?

Up to discussion.

Reply:
> I didn't implement the tests like test_feedback_no_payload_send. Those are basically testing the FastAPI/pydantic behavior. Unless there is a special logic for handling/validating the input parameters, I'm not sure what is the actual benefit of such tests.

there are two benefits to writing such tests, imho:

1) writing the test forces us to write down(in the test) what we expect the failure behavior to be, which can lead to a useful discussion about whether that failure behavior is what we want it to be(e.g. is the error code we want? is there a useful error message associated that will help someone understand what went wrong?).  In general tests(especially integration and e2e tests) are a way for us to express the expected behavior of the system

2) if we accidentally break things such that an empty payload does *not* result in failure, that would be bad.  So the point of the test is to ensure that we fail when we expect to fail, because if we don't fail when we should have, that's a problem.

I am not going to claim they are the most important tests, but i think they do provide value.